ceph/rados/core_workflows.py

```diff
@@ -4615,3 +4623,57 @@ def get_rados_df(self, pool_name: str = None):
         out = self.run_ceph_command(cmd=_cmd, client_exec=True)
 
         return out["pools"][0] if pool_name else out
+
+    def set_service_managed_type(self, service_type, unmanaged) -> bool:
+        """
+        Method to set the service to either managed or unmanaged
+        Args:
+            unmanaged: True or false, for the service management
+            service_type : service types are- mon,mgr,osd,rgw, mds
+        returns:
+            Pass -> True, Fail -> false
+        """
+        file_name = (
+            f"/tmp/{service_type}_spec_{self.set_service_managed_type.__name__}.yaml"
+        )
+
+        # Creating service config file
+        self.client.exec_command(sudo=True, cmd=f"touch {file_name}")
+
+        cmd_export = f"ceph orch ls {service_type} --export"
+        out = self.run_ceph_command(cmd=cmd_export, client_exec=True)
+        for osd_service in out:
+            if unmanaged:
+                log.debug(
+                    f"Setting the {service_type} service as unmanaged by cephadm. current status : {out}"
+                )
+                osd_service["unmanaged"] = "true"
+            else:
+                log.debug(
+                    f"Setting the {service_type} service as unmanaged by cephadm. current status : {out}"
+                )
+                osd_service["unmanaged"] = "false"
+            json_out = json.dumps(osd_service)
+            # Adding the spec rules into the file
+            cmd = f"echo '{json_out}' > {file_name}"
+            self.client.exec_command(cmd=cmd, sudo=True)
+            log.debug(f"Contents of {service_type} spec file : {out}")
+            apply_cmd = f"ceph orch apply -i {file_name}"
+            log.info(f"Applying the spec file via cmd : {apply_cmd}")
+            self.client.exec_command(cmd=apply_cmd, sudo=True)
+            time.sleep(10)
+            out = self.list_orch_services(service_type=service_type, export=True)
```

Comment:
this method call should be outside the for loop

Reply:
Moved outside the loop